### DISCUSSION TITLE
Shouldn't we steal treesitter textobjects from the neovim plugin?

### DISCUSSION THREAD

**Discussion by k2d222** [helix-editor/helix#1404] [Ideas] [answered]
The popular neovim plugin `nvim-treesitter-textobjects` has textobjects queries mostly compatible with helix's (see [https://github.com/nvim-treesitter/nvim-treesitter-textobjects/tree/master/queries](https://github.com/nvim-treesitter/nvim-treesitter-textobjects/tree/master/queries)).

Only difference is helix uses `inside` and `around` instead of `inner` and `outer`. Replacing those, they seem to work properly (I tested js/ts and ocaml). And they are more detailed, paving the way for eventual extensions. Would make sense to use those and not reinvent the wheel.

I could be missing a licence issue? I don't know much about that.

note: If we do use those queries, I believe it would be more clever to adapt our syntax to use `inner` and `outer` as I believe it is in our interest to follow "standards" when it doesn't cost anything.

**Comment by dead10ck** [comment, marked as answer]
It's my understanding that many of the queries in helix are based on these ones, because like you said, why reinvent the wheel. I think the reason we don't just reuse the repo outright is because of the capture name differences, and also I believe the order of precedence is backwards in neovim: tree-sitter and helix favor mosr recent first, and neovim prefers the first definition. This leads to some incorrect highlighting without fixing.
